### ISSUE TITLE
Accessibility: Calendar does not show when focusing the input with keyboard navigation after it has been closed with Escape

### ISSUE THREAD

**Issue by mgarciamartin** [Hacker0x01/react-datepicker#2146]
**Describe the bug**
The calendar popup does not show when focusing the input with keyboard navigation if it has been closed with Escape previously.
It works fine the first time when focusing the input with keyboard navigation. After closing the calendar with Escape it does not work anymore.

**To Reproduce**
1. Go to sandbox https://codesandbox.io/s/jovial-bassi-y8kg9?file=/src/app.js
2. Click on the preview body to focus it.
3. Hit Tab until the Datepicker is focused and the calendar is shown.
4. Press Escape to close the calendar.
5. Click on the preview body to focus it.
6. Hit Tab until the Datepicker is focused.
7. The calendar is not shown.

**Expected behavior**
Everytime the Datepicker gains focus through keyboard navigation the calendar should be shown.

**Additional context**
Version of the library: 2.15.0
